src/openlcb/DccAccyProducer.cxxtest

```diff
@@ -0,0 +1,274 @@
+#include "utils/async_if_test_helper.hxx"
+
+#include "DccAccyProducer.hxx"
+
+namespace openlcb
+{
+
+using namespace std::placeholders;
+
+class DccAccyProducerTest : public AsyncNodeTest
+{
+public:
+    MOCK_METHOD2(state_callback1, void(unsigned, bool));
+    MOCK_METHOD2(state_callback2, void(unsigned, bool));
+
+protected:
+    DccAccyProducerTest()
+    {
+        print_all_packets();
+        wait();
+    }
+
+public:
+    SyncNotifiable n_;
+
+    class ImplFlow : public StateFlowBase
+    {
+    public:
+        ImplFlow(DccAccyProducerTest *test)
+            : StateFlowBase(test->ifCan_.get())
+            , dap1_(test->node_,
+                 std::bind(&DccAccyProducerTest::state_callback1, test, _1, _2))
+            , dap2_(test->node_,
+                 std::bind(&DccAccyProducerTest::state_callback2, test, _1, _2))
+            , test_(test)
+        {
+        }
+
+        ~ImplFlow()
+        {
+            auto *b1 = full_allocation_result(&dap1_);
+            auto *b2 = full_allocation_result(&dap2_);
+            if (b1)
+            {
+                b1->unref();
+            }
+            if (b2)
+            {
+                b2->unref();
+            }
+        }
+
+        DccAccyProducerInput *response1()
+        {
+            return full_allocation_result(&dap1_)->data();
+        }
+
+        DccAccyProducerInput *response2()
+        {
+            return full_allocation_result(&dap2_)->data();
+        }
+
+        DccAccyProducer dap1_;
+        DccAccyProducer dap2_;
+
+        DccAccyProducerTest *test_;
+    };
+
+};
+
+TEST_F(DccAccyProducerTest, SendSet)
+{
+    wait();
+    clear_expect();
+
+    class TestFlow : public ImplFlow
```

Comment:
If you don't want to do this explicit but complicated way of testing, there is a helper function for invoking subflows that you can use: https://github.com/bakerstu/openmrn/blob/master/src/utils/async_if_test_helper.hxx#L288

That you can run directly inside the TEST_F function body and will handle the invocation, the wait for return, and the ownership of the returned pointer.

Look at openlcb/TractionThrottle.cxxtest:254 for examples.